06_predict_survival/nbconverted/plot_survival_results.py

```diff
@@ -0,0 +1,207 @@
+):
+    
+    ax = axarr[ix]
+    
+    filtered_df = results_df[
+        (results_df.training_data == data_type) &
+        (results_df.data_type == 'test')
+    ].copy()
+
+    filtered_df.sort_values(by='cancer_type', inplace=True)
+
+    sns.boxplot(data=filtered_df, x='cancer_type', y='cindex', hue='signal',
+                hue_order=['signal', 'shuffled'], ax=ax)
+    ax.set_xlabel('TCGA cancer type')
+    ax.set_ylabel('Concordance index')
+    ax.set_title('Survival prediction using {}, top 1000 predictors by MAD'.format(data_type))
+    ax.set_ylim(0.0, 1.0)
+    
+plt.tight_layout()
+
+
+# In[7]:
+
+
+results_df['identifier'] = (results_df.cancer_type + '_' +
+                            results_df.training_data)
+results_df.head()
+
+
+# In[8]:
+
+
+compare_df = au.compare_control_ind(results_df,
+                                    identifier='identifier',
+                                    metric='cindex',
+                                    verbose=True)
+compare_df['cancer_type'] = compare_df.identifier.str.split('_', 1, expand=True)[0]
+compare_df['training_data'] = compare_df.identifier.str.split('_', 1, expand=True)[1]
+
+print(len(compare_df))
+compare_df.head()
+
+
+# In[9]:
+
+
+sns.set({'figure.figsize': (18, 8)})
+    
+compare_df.sort_values(by='cancer_type', inplace=True)
+
+sns.boxplot(data=compare_df[~compare_df.cancer_type.isin(drop_cancer_types)],
+            x='cancer_type',
+            y='delta_cindex',
+            hue='training_data',
+            hue_order=['expression', 'me_27k', 'me_450k'])
+plt.xlabel('TCGA cancer type')
+plt.ylabel('cindex(signal) - cindex(shuffled)')
+plt.title('Survival prediction, top 1000 predictors by MAD')
+plt.ylim(-0.5, 0.5)
+    
+plt.tight_layout()
+
+
+# ### Heatmap
+# 
+# Heatmap
+
+# In[10]:
+
+
+all_results_df = au.compare_all_data_types(results_df[~results_df.cancer_type.isin(drop_cancer_types)],
+                                           SIG_ALPHA,
+                                           identifier='cancer_type',
+                                           metric='cindex')
+
+all_results_df.rename(columns={'gene': 'cancer_type'}, inplace=True)
+all_results_df.sort_values(by='p_value').head(10)
+
+
+# In[11]:
+
+
+heatmap_df = (all_results_df
+    .pivot(index='training_data', columns='cancer_type', values='delta_mean')
+    .reindex(['expression', 'me_27k', 'me_450k'])
+)
+heatmap_df.iloc[:, :5]
+
+
+# In[12]:
+
+
+raw_results_df = (results_df
+    .drop(columns=['identifier'])
+    .rename(columns={'cancer_type': 'identifier'})
+)
+raw_results_df.head()
+
+
+# In[13]:
+
+
+sns.set({'figure.figsize': (28, 6)})
+sns.set_context('notebook', font_scale=1.5)
+
+ax = plu.plot_heatmap(heatmap_df,
+                      all_results_df.reset_index(drop=True),
+                      different_from_best=True,
+                      raw_results_df=raw_results_df,
+                      metric='cindex',
+                      id_name='cancer_type')
```

Comment:
Transcriptomics = best omics

I know previously you found that a model trained on a concatentation of 'omics performed <= just expression. Would you expect the same to be true here? Do you have an intuition for why expression and methylation seem to be conveying the same information?

Reply:
Personally, I would expect it to be the same here, although I don't think we're going to try the multi-omics models for survival prediction right now (we want to get this paper submitted soon and it already has a ton of data/figures in it).

I think the reasons could be either biological (driver mutations with strong roles in cancer development lead to everything getting messed up, resulting in signatures of cancer dysregulation in both DNA methylation/epigenetics and gene expression patterns; also promoter methylation affects gene expression so we'd expect to see a bit of a causal link) or technical (we're using a really simple multi-omics method of just concatenating datasets, and it's possible that more sophisticated methods would align the signal better/extract the unique information more effectively in each data layer).

Distinguishing between the two is definitely a topic we're interested in for future work, but probably beyond the scope of this paper.